the-collab-lab/tcl-25-smart-shopping-list
# List view styling

## Description

This PR removes the background color on the whole list item and adds it on the checkbox only. It also categorizes the items in different groups: soon, kind of soon, inactive, and not soon. 

## Related Issue
#31 


## Acceptance Criteria
- [x] Add a background color to the checkbox only
- [x] Group the items into categories based on how soon they are likely to be bought
- [x] Add a title to each category
- [x] Use appropriate styling for the titles and item names



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|   ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |



## Testing Steps / QA Criteria

1. From your terminal pull down this branch with `git pull list-view-styling` and check that branch out with `git checkout list-view-styling.`
2. Then `npm start` to launch the app.
3. Create a `new shopping list `or` join with an existing token.`
4. Add items using the add view. Ensure that the shopping list has at least 4 items. 
6. Find a document in the [database](https://console.firebase.google.com/u/1/project/tcl-25-shopping-list/firestore/data~2Flists) with your token.
7.  Make the first item `DaysLeftForTheNextPurchaseDate` 's value to be bellow 7 and `numberOfPurchases`'s to be above 2.
8. Make the second item `DaysLeftForTheNextPurchaseDate` 's value to be between 7 and 30 and `numberOfPurchases`'s to be above 2.
9. Make the third item `DaysLeftForTheNextPurchaseDate` 's value to be above 30 and `numberOfPurchases`'s to be above  2.
10. Make the fourth item `numberOfPurchases`  's value to be bellow 2.
11. Return to the app's home screen. Check out the list view to see how the four items are organized into groups, each with a different checkbox background colour.
